src/sentry/utils/snuba.py

```diff
@@ -126,6 +126,12 @@ def raw_query(start, end, groupby=None, conditions=None, filter_keys=None,
     with timer('get_project_issues'):
         issues = get_project_issues(project_ids, filter_keys.get('issue')) if get_issues else None
 
+    if len(issues) == 1:
```

Comment:
I figured we'd start with Snuba queries against a single issue, because:
* They are common, especially for tagstore queries when you view a single issue.
* We have to query for the `Group`(s), because it'd be a huge refactor to have every Snuba caller pass down actual `Group` objects (that they may or may not have). I figure we should prove the idea out first.
* (Least important) I would assume groups of 500 issues would often blow out to full retention anyway, but we can easily add it if we desire.